### DISCUSSION TITLE
Remove old intel binaries

### DISCUSSION THREAD

**Discussion by Hipska** [Homebrew/discussions#4307] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 4d0154fa7926e8a7b06e9b074dd45957e820407c
Last commit: 7 days ago
Core tap JSON: 06 Mar 09:29 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.21AxeQAe2A/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_blizzard_avalanche
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libbeid.2.7.4.dylib
  /usr/local/lib/libbeidcardlayer.4.0.7.dylib
  /usr/local/lib/libbeidcommon.1.0.0.dylib
  /usr/local/lib/libbeidcommon.4.0.7.dylib
  /usr/local/lib/libbeiddialogsQT.4.0.7.dylib
  /usr/local/lib/libbeidgui.1.5.0.dylib
  /usr/local/lib/libbeidlibjni.2.7.2.dylib
  /usr/local/lib/libbeidlibopensc.2.2.0.dylib
  /usr/local/lib/libbeidlibopenscinternal.2.2.0.dylib
  /usr/local/lib/libbeidpkcs11.4.0.7.dylib
  /usr/local/lib/libbeidpkcs11.4.1.11.dylib
  /usr/local/lib/libbeidpkcs11.4.2.7.dylib
  /usr/local/lib/libqt-mt.3.3.8.dylib
```

### Description of issue

I migrated from an intel mac to an M2, and because of that all the old brewed binaries are still linked in `/usr/local`. I'm trying to run all my software without rosetta, so that's not installed.

Now I would like to be able to unlink all formula's in `/usr/local` and stay with only the ones in `/opt/homebrew`. If possible I would also like to clean the intel based Cellar directory.

I tried with for example the following command, but that's not doing what I want: 
```
$ HOMEBREW_PREFIX=/usr/local brew unlink php -n
Would remove:
/opt/homebrew/bin/pear
/opt/homebrew/bin/peardev
/opt/homebrew/bin/pecl
/opt/homebrew/bin/phar
/opt/homebrew/bin/phar.phar
/opt/homebrew/bin/php
/opt/homebrew/bin/php-cgi
/opt/homebrew/bin/php-config
/opt/homebrew/bin/phpdbg
/opt/homebrew/bin/phpize
/opt/homebrew/include/php
/opt/homebrew/lib/httpd
/opt/homebrew/lib/php/20220829
/opt/homebrew/lib/php/build
/opt/homebrew/sbin/php-fpm
/opt/homebrew/share/man/man1/phar.1
/opt/homebrew/share/man/man1/phar.phar.1
/opt/homebrew/share/man/man1/php-cgi.1
/opt/homebrew/share/man/man1/php-config.1
/opt/homebrew/share/man/man1/php.1
/opt/homebrew/share/man/man1/phpdbg.1
/opt/homebrew/share/man/man1/phpize.1
/opt/homebrew/share/man/man8/php-fpm.8
/opt/homebrew/share/php
```

**Comment by SMillerDev** [comment, marked as answer]
Without Rosetta installed brew won't manage the `/usr/local` directory. If you don't have rosetta you can't really use the `/usr/local` install though, so you might as well just `rm -rf /usr/local/Cellar/php` when you moved what you want to move.

**Reply by Hipska** [reply]
Yes I know, I'm just looking at the best/easiest way to remove all the unneeded clutter. As I would need to look into bin, include, lib, sbin, share and possibly many other directories to remove all broken symlinks if I just remove the files in Cellar..

**Reply by SMillerDev** [reply]
But the symlinks don't matter because you're not running anything from there, right? And you can just remove those entire directories as soon as you've extracted everything from the cellar.

**Reply by Hipska** [reply]
`/usr/local` is in the PATH for many applications/tools, right? I'm indeed not specifically using these symlinks, but 'some' tools might and I want to avoid that.

Anyway, you are saying I can safely just remove everything under `/usr/local/Cellar`?

**Reply by SMillerDev** [reply]
> `/usr/local` is in the PATH for many applications/tools, right?

Some install there, but only Homebrew would install and symlink to Cellar.

> Anyway, you are saying I can safely just remove everything under /usr/local/Cellar?

Yes, apart from configuration and data everything there is x86 specific so it'll be useless to you.

**Reply by Hipska** [reply]
Thanks, then I only need to find something to remove all dangling symlinks in `/usr/local/*`..

Edit, used this: `find /usr/local -type l -ls | grep Cellar | awk '{print $(NF-2)}' | xargs rm -v --`

**Reply by gromgit** [reply]
> remove all dangling symlinks in `/usr/local/*`

`find -L /usr/local -type l -exec rm -v -- {} +`
